File: coding/src/main/scala/org/shev4ik/coding/sorting/SelectionSort.scala
package org.shev4ik.coding.sorting

object SelectionSort extends App {
  val arr = Array(3, 7, 9, 8, 4, 5, 1, 6, 2, 0)

  sort(arr) foreach print
  println("")
  sortRecursion(arr) foreach print

  def sort(arr: Array[Int]): Array[Int] = {
    for (i <- arr.indices) {
      var min = Integer.MAX_VALUE
      var ind = -1
      for (j <- i until arr.length) {
        if (min > arr(j)) {
          min = arr(j)
          ind = j
        }
      }
      val temp = arr(i)
      arr(i) = min
      arr(ind) = temp
    }
    arr
  }

  def sortRecursion(arr: Array[Int]): Array[Int] = {
    def inner(l: List[Int], result: List[Int]): List[Int] = {
      l match {
        case Nil ⇒ result
        case list: List[Any] ⇒
          val m      = list.min
          val (h, t) = list.span(_ != m)
          inner(h ++ t.tail, result :+ m)
      }
    }
    inner(arr.toList, Nil).toArray
  }
}

File: coding/src/main/kotlin/org.kotlin.algorithms/SelectionSort.kt
package org.kotlin.algorithms

fun main(){

    val arr = arrayOf(3,5,2,1,6,7,4)

    for(i in arr.indices){
        var minIndex = i
        for(j in (i+1).until(arr.size)){
            if (arr[j] < arr[minIndex]) {
                minIndex = j
            }
        }
        val temp = arr[i]
        arr[i] = arr[minIndex]
        arr[minIndex] = temp
    }


    for(index in arr.indices){
        println(arr[index])
    }


}